ssl/ssl_locl.h

```diff
@@ -384,6 +384,14 @@
 # define SSL_CLIENT_USE_SIGALGS(s)        \
     SSL_CLIENT_USE_TLS1_2_CIPHERS(s)
 
+# define IS_MAX_FRAGMENT_LENGTH_EXT_VALID(value) \
+    (((value) >= TLSEXT_max_fragment_length_2_TO_9) && \
+     ((value) <= TLSEXT_max_fragment_length_2_TO_12))
+# define USE_MAX_FRAGMENT_LENGTH_EXT(session) \
+    IS_MAX_FRAGMENT_LENGTH_EXT_VALID(session->ext.max_fragment_length)
+# define GET_MAX_FRAGMENT_LENGTH(session) \
+    (2U << (7 + session->ext.max_fragment_length))
```

Comment:
I found this really confusing. Why are you shifting 2? I'd either shift 1 or 512, i.e. either

    (1U << (8 + session->ext.max_fragment_length))

or (perhaps better)

    (512U << (session->ext.max_fragment_length - 1))